linera-views/DESIGN.md

```diff
@@ -0,0 +1,140 @@
+This document collects some design choices that were made in the `linera-views`.
+For a more basic introduction, see `README.md`.
+
+# Design
+
+## Key Value Store Clients
+
+We have designed a `KeyValueStoreClient` trait that represents the basic functionalities
+of a key-value store whose keys are `Vec<u8>` and whose values are `Vec<u8>`.
+
+We provide an implementation of the trait `KeyValueStoreClient` for the following key-value stores:
+* `MemoryClient` is using the memory (and uses internally a simple B-Tree map).
+* `RocksdbClient` is a disk-based key-value store
+* `DynamoDbClient` is the AWS-based DynamoDB service.
+
+The trait `KeyValueStoreClient` was designed so that more storage solutions can be easily added in the future.
+
+The `KeyValueStoreClient` trait is also implemented for several internal constructions of clients:
+* The `LruCachingKeyValueClient` client implements the Least Recently Used (LRU)
```

Comment:
```suggestion
* The `LruCachingKeyValueClient<K>` client implements the Least Recently Used (LRU)
```

(Or we could omit the generic parameters from the other two?)

Reply:
Ok, done.